ssr/src/component/auth_providers/mod.rs

```diff
@@ -137,6 +137,10 @@ pub fn LoginProviders(show_modal: RwSignal<bool>, lock_closing: RwSignal<bool>)
                 log::warn!("failed to handle user login, err {e}. skipping");
             }
 
+            let (_, set_user_canister_id, _) =
+                use_local_storage::<Option<Principal>, JsonSerdeCodec>(USER_CANISTER_ID);
+            set_user_canister_id(Some(canisters.user_canister()));
+
```

Comment:
You don't need to redo this here,
[This](https://github.com/go-bazzinga/hot-or-not-web-leptos-ssr/blob/9bb297823b3f56cfb2835c092f7c7a9994d90525/ssr/src/component/auth_providers/mod.rs#L159) will retrigger `canisters_res` and base_route will set the appropriate canister